src/dmd/mars.d

```diff
@@ -1869,6 +1880,11 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
                         goto Lnoarg;
                     params.jsonfilename = p + 3 + (p[3] == '=');
                     break;
+                case 'q':
+                    if (!p[3])
+                        goto Lnoarg;
+                    params.jsonQuery = p + 3 + (p[3] == '=');
```

Comment:
check p[3] == '=' and `params.jsonQuery = p + 4`
We really should not allow things like `-Xqfoo` in new flags; this is old baggage from dmd cmd line (eg `-offile` instead of (`-of=file`), the downsides have these have been discussed: it's not standard, hard to read visually, and disallows future new flags (eg `-Xquery=baz`)

Reply:
Yes I agree with this, I'll remove support for omitting `=`.